### PR TITLE
[HZ-710] Introduce CREATE VIEW statement

### PR DESCRIPTION
**Pull request by Fly-Style**
This is first PR for VIEW feature, and it introduces syntax and backend support for `CREATE VIEW` and `DROP VIEW` SQL statements.

Query substitution will be implemented in further PR.

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set

### CODE REVIEW

**Comment by Fly-Style** [comment]
Note: I am planning to create Hazelcast SQL dialect ([link to current solution](https://github.com/hazelcast/hazelcast/pull/19894/files#diff-e37dd5dee19ecff52d9c17b5d57023de81923b4f52ac39bb0cddce736f2aff40R381))

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/ParserResource.java [hunk 1]
```diff
@@ -34,6 +34,9 @@
     @BaseMessage("The mapping must be created in the \"public\" schema")
     ExInst<SqlValidatorException> mappingIncorrectSchema();
 
+    @BaseMessage("The view must be created in the \"public\" schema")
```
[the remark this answers is not in the document]
Comment (Fly-Style) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/ParserResource.java:37]: It's a message user will see. I assume, user would see more specific error message...

**Comment by Fly-Style** [comment]
> This combination actually isn't allowed, it's an implementation detail that the method allows it. Should we test it?

@viliam-durina I provided [a tests](https://github.com/hazelcast/hazelcast/pull/19894/files?authenticity_token=fdNvt1sWOx2hRx1h%2BxnbYKfOxDTGYoinUN0zIzB3fasY9L7qH1zFk6BOg9fFMlVGcjDuaW5wTzFCDsQ1ndUpJQ%3D%3D&file-filters%5B%5D=.java#diff-93c93c5ad23a6f9e409d2f3305a707defdf828061f00b3e542d2eb3c10e2ad70R105) that show that this combination is forbidden.

**Review by viliam-durina — APPROVED** [review]

**Review by ivanthescientist — APPROVED** [review]

**Comment by ivanthescientist** [comment]
run-lab-run

**Pull request merged**